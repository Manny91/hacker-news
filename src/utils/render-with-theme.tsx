import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "../styled-components";
import { theme } from "../theme";
import "jest-styled-components";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
