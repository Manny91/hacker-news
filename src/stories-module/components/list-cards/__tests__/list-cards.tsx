import React from "react";
import { renderWithTheme } from "../../../../utils/render-with-theme";
import ListCards from "../list-cards";

describe("ListCards Component", () => {
  it("should render properly", () => {
    const subject = renderWithTheme(<ListCards />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
