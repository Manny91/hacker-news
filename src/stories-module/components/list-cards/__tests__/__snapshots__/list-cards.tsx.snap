// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCards Component should render properly 1`] = `
.c1 {
  margin-right: 40px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  list-style-type: none;
}

.c1:last-child {
  margin-right: 0px;
}

.c13 {
  background-color: #14137d;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #f6f6f6;
  border-radius: 7px;
}

.c12 {
  height: 18px;
}

.c4 {
  font-size: 20px;
}

.c15 {
  font-size: 20px;
}

.c14 {
  font-size: 20px;
}

.c3 {
  background-color: #14137d;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #f6f6f6;
  border-radius: 7px;
  background-color: #00ffff;
}

.c11 {
  background-color: #14137d;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #f6f6f6;
  border-radius: 7px;
  background-color: #4285f4;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 15px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  margin: 0px;
  font-weight: bold;
  font-size: 22px;
  line-height: normal;
  color: #020202;
}

.c10 {
  margin: 0px;
  font-weight: bold;
  font-size: 22px;
  line-height: normal;
  color: #020202;
  text-transform: uppercase;
  font-weight: 400;
  font-size: 11px;
  color: grey;
}

.c9 {
  height: 10px;
}

.c8 {
  margin-left: 5px;
}

.c2 {
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 7px;
  height: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px;
  width: 100%;
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-right: 0px;
    margin-bottom: 20px;
  }

  .c1:last-child {
    margin-bottom: 0px;
  }
}

@media (max-width:992px) {
  .c13 {
    display: none;
  }
}

@media (max-width:992px) {
  .c3 {
    display: none;
  }
}

@media (max-width:992px) {
  .c11 {
    display: none;
  }
}

@media (max-width:992px) {
  .c5 {
    padding: 7px;
  }
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <span
            aria-label="line-chart"
            class="anticon anticon-line-chart c4"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="line-chart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM305.8 637.7c3.1 3.1 8.1 3.1 11.3 0l138.3-137.6L583 628.5c3.1 3.1 8.2 3.1 11.3 0l275.4-275.3c3.1-3.1 3.1-8.2 0-11.3l-39.6-39.6a8.03 8.03 0 00-11.3 0l-230 229.9L461.4 404a8.03 8.03 0 00-11.3 0L266.3 586.7a8.03 8.03 0 000 11.3l39.5 39.7z"
              />
            </svg>
          </span>
        </div>
        <div
          class="c5"
        >
          <div>
            <div
              class="c6"
            >
              <p
                class="c7"
              >
                $510
              </p>
              <div
                class="c8"
                style="color: rgb(56, 158, 13);"
              >
                <span
                  aria-label="arrow-up"
                  class="anticon anticon-arrow-up c9"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="arrow-up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M868 545.5L536.1 163a31.96 31.96 0 00-48.3 0L156 545.5a7.97 7.97 0 006 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
                    />
                  </svg>
                </span>
                <b>
                  50
                  %
                </b>
              </div>
            </div>
            <p
              class="c10"
            >
              revenue
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c11"
        >
          <svg
            class="c12"
          >
            zap.svg
          </svg>
        </div>
        <div
          class="c5"
        >
          <div>
            <div
              class="c6"
            >
              <p
                class="c7"
              >
                12.2M
              </p>
              <div
                class="c8"
                style="color: rgb(56, 158, 13);"
              >
                <span
                  aria-label="arrow-up"
                  class="anticon anticon-arrow-up c9"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="arrow-up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M868 545.5L536.1 163a31.96 31.96 0 00-48.3 0L156 545.5a7.97 7.97 0 006 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
                    />
                  </svg>
                </span>
                <b>
                  12
                  %
                </b>
              </div>
            </div>
            <p
              class="c10"
            >
              impressions
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c13"
        >
          <span
            aria-label="global"
            class="anticon anticon-global c14"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="global"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M854.4 800.9c.2-.3.5-.6.7-.9C920.6 722.1 960 621.7 960 512s-39.4-210.1-104.8-288c-.2-.3-.5-.5-.7-.8-1.1-1.3-2.1-2.5-3.2-3.7-.4-.5-.8-.9-1.2-1.4l-4.1-4.7-.1-.1c-1.5-1.7-3.1-3.4-4.6-5.1l-.1-.1c-3.2-3.4-6.4-6.8-9.7-10.1l-.1-.1-4.8-4.8-.3-.3c-1.5-1.5-3-2.9-4.5-4.3-.5-.5-1-1-1.6-1.5-1-1-2-1.9-3-2.8-.3-.3-.7-.6-1-1C736.4 109.2 629.5 64 512 64s-224.4 45.2-304.3 119.2c-.3.3-.7.6-1 1-1 .9-2 1.9-3 2.9-.5.5-1 1-1.6 1.5-1.5 1.4-3 2.9-4.5 4.3l-.3.3-4.8 4.8-.1.1c-3.3 3.3-6.5 6.7-9.7 10.1l-.1.1c-1.6 1.7-3.1 3.4-4.6 5.1l-.1.1c-1.4 1.5-2.8 3.1-4.1 4.7-.4.5-.8.9-1.2 1.4-1.1 1.2-2.1 2.5-3.2 3.7-.2.3-.5.5-.7.8C103.4 301.9 64 402.3 64 512s39.4 210.1 104.8 288c.2.3.5.6.7.9l3.1 3.7c.4.5.8.9 1.2 1.4l4.1 4.7c0 .1.1.1.1.2 1.5 1.7 3 3.4 4.6 5l.1.1c3.2 3.4 6.4 6.8 9.6 10.1l.1.1c1.6 1.6 3.1 3.2 4.7 4.7l.3.3c3.3 3.3 6.7 6.5 10.1 9.6 80.1 74 187 119.2 304.5 119.2s224.4-45.2 304.3-119.2a300 300 0 0010-9.6l.3-.3c1.6-1.6 3.2-3.1 4.7-4.7l.1-.1c3.3-3.3 6.5-6.7 9.6-10.1l.1-.1c1.5-1.7 3.1-3.3 4.6-5 0-.1.1-.1.1-.2 1.4-1.5 2.8-3.1 4.1-4.7.4-.5.8-.9 1.2-1.4a99 99 0 003.3-3.7zm4.1-142.6c-13.8 32.6-32 62.8-54.2 90.2a444.07 444.07 0 00-81.5-55.9c11.6-46.9 18.8-98.4 20.7-152.6H887c-3 40.9-12.6 80.6-28.5 118.3zM887 484H743.5c-1.9-54.2-9.1-105.7-20.7-152.6 29.3-15.6 56.6-34.4 81.5-55.9A373.86 373.86 0 01887 484zM658.3 165.5c39.7 16.8 75.8 40 107.6 69.2a394.72 394.72 0 01-59.4 41.8c-15.7-45-35.8-84.1-59.2-115.4 3.7 1.4 7.4 2.9 11 4.4zm-90.6 700.6c-9.2 7.2-18.4 12.7-27.7 16.4V697a389.1 389.1 0 01115.7 26.2c-8.3 24.6-17.9 47.3-29 67.8-17.4 32.4-37.8 58.3-59 75.1zm59-633.1c11 20.6 20.7 43.3 29 67.8A389.1 389.1 0 01540 327V141.6c9.2 3.7 18.5 9.1 27.7 16.4 21.2 16.7 41.6 42.6 59 75zM540 640.9V540h147.5c-1.6 44.2-7.1 87.1-16.3 127.8l-.3 1.2A445.02 445.02 0 00540 640.9zm0-156.9V383.1c45.8-2.8 89.8-12.5 130.9-28.1l.3 1.2c9.2 40.7 14.7 83.5 16.3 127.8H540zm-56 56v100.9c-45.8 2.8-89.8 12.5-130.9 28.1l-.3-1.2c-9.2-40.7-14.7-83.5-16.3-127.8H484zm-147.5-56c1.6-44.2 7.1-87.1 16.3-127.8l.3-1.2c41.1 15.6 85 25.3 130.9 28.1V484H336.5zM484 697v185.4c-9.2-3.7-18.5-9.1-27.7-16.4-21.2-16.7-41.7-42.7-59.1-75.1-11-20.6-20.7-43.3-29-67.8 37.2-14.6 75.9-23.3 115.8-26.1zm0-370a389.1 389.1 0 01-115.7-26.2c8.3-24.6 17.9-47.3 29-67.8 17.4-32.4 37.8-58.4 59.1-75.1 9.2-7.2 18.4-12.7 27.7-16.4V327zM365.7 165.5c3.7-1.5 7.3-3 11-4.4-23.4 31.3-43.5 70.4-59.2 115.4-21-12-40.9-26-59.4-41.8 31.8-29.2 67.9-52.4 107.6-69.2zM165.5 365.7c13.8-32.6 32-62.8 54.2-90.2 24.9 21.5 52.2 40.3 81.5 55.9-11.6 46.9-18.8 98.4-20.7 152.6H137c3-40.9 12.6-80.6 28.5-118.3zM137 540h143.5c1.9 54.2 9.1 105.7 20.7 152.6a444.07 444.07 0 00-81.5 55.9A373.86 373.86 0 01137 540zm228.7 318.5c-39.7-16.8-75.8-40-107.6-69.2 18.5-15.8 38.4-29.7 59.4-41.8 15.7 45 35.8 84.1 59.2 115.4-3.7-1.4-7.4-2.9-11-4.4zm292.6 0c-3.7 1.5-7.3 3-11 4.4 23.4-31.3 43.5-70.4 59.2-115.4 21 12 40.9 26 59.4 41.8a373.81 373.81 0 01-107.6 69.2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="c5"
        >
          <div>
            <div
              class="c6"
            >
              <p
                class="c7"
              >
                37.5%
              </p>
              <div
                class="c8"
                style="color: rgb(207, 19, 34);"
              >
                <span
                  aria-label="arrow-up"
                  class="anticon anticon-arrow-up c9"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="arrow-up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M868 545.5L536.1 163a31.96 31.96 0 00-48.3 0L156 545.5a7.97 7.97 0 006 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
                    />
                  </svg>
                </span>
                <b>
                  12
                  %
                </b>
              </div>
            </div>
            <p
              class="c10"
            >
              fill-rate
            </p>
          </div>
        </div>
      </div>
    </li>
    <li
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c13"
        >
          <span
            aria-label="deployment-unit"
            class="anticon anticon-deployment-unit c15"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="deployment-unit"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M888.3 693.2c-42.5-24.6-94.3-18-129.2 12.8l-53-30.7V523.6c0-15.7-8.4-30.3-22-38.1l-136-78.3v-67.1c44.2-15 76-56.8 76-106.1 0-61.9-50.1-112-112-112s-112 50.1-112 112c0 49.3 31.8 91.1 76 106.1v67.1l-136 78.3c-13.6 7.8-22 22.4-22 38.1v151.6l-53 30.7c-34.9-30.8-86.8-37.4-129.2-12.8-53.5 31-71.7 99.4-41 152.9 30.8 53.5 98.9 71.9 152.2 41 42.5-24.6 62.7-73 53.6-118.8l48.7-28.3 140.6 81c6.8 3.9 14.4 5.9 22 5.9s15.2-2 22-5.9L674.5 740l48.7 28.3c-9.1 45.7 11.2 94.2 53.6 118.8 53.3 30.9 121.5 12.6 152.2-41 30.8-53.6 12.6-122-40.7-152.9zm-673 138.4a47.6 47.6 0 01-65.2-17.6c-13.2-22.9-5.4-52.3 17.5-65.5a47.6 47.6 0 0165.2 17.6c13.2 22.9 5.4 52.3-17.5 65.5zM522 463.8zM464 234a48.01 48.01 0 0196 0 48.01 48.01 0 01-96 0zm170 446.2l-122 70.3-122-70.3V539.8l122-70.3 122 70.3v140.4zm239.9 133.9c-13.2 22.9-42.4 30.8-65.2 17.6-22.8-13.2-30.7-42.6-17.5-65.5s42.4-30.8 65.2-17.6c22.9 13.2 30.7 42.5 17.5 65.5z"
              />
            </svg>
          </span>
        </div>
        <div
          class="c5"
        >
          <div>
            <div
              class="c6"
            >
              <p
                class="c7"
              >
                $2.4
              </p>
              <div
                class="c8"
                style="color: rgb(56, 158, 13);"
              >
                <span
                  aria-label="arrow-up"
                  class="anticon anticon-arrow-up c9"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="arrow-up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M868 545.5L536.1 163a31.96 31.96 0 00-48.3 0L156 545.5a7.97 7.97 0 006 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
                    />
                  </svg>
                </span>
                <b>
                  32
                  %
                </b>
              </div>
            </div>
            <p
              class="c10"
            >
              ecpm
            </p>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
